### ISSUE TITLE
java.home shouldn't be configurable on Folder level

### ISSUE THREAD

**Issue by anthonyvdotbe** [redhat-developer/vscode-java#1769]
Maybe I'm missing something, but I don't see how it makes sense to be able to configure `java.home` on the Folder level

##### Environment
- Operating System: Windows 10
- JDK version: 15.0.1
- Visual Studio Code version: 1.52.1
- Java extension version: 0.73.0

##### Steps To Reproduce
1. open Folder-specific settings & search for java.home

##### Current Result
java.home isn't configurable

##### Expected Result
java.home is configurable

##### Additional Informations
![image](https://user-images.githubusercontent.com/15714253/104808165-a954d080-57e4-11eb-988c-180fa2dee7bb.png)

**Comment by testforstephen** [comment]
At the earlier version of Java extension, we set "java.home" to `window` scope, which is configurable in user, workspace, remote level. But we found this didn't work for remote scenarios, because "java.home" points to an absolute path of local machine, it is not shareable for remote machine. We can only choose between `machine` and `machine-overridable`. In order not to break the existing users who have configured "java.home" in workspace settings, we end up choosing `machine-overridable`. 

Since "java.home" is just used to launch the Java tooling itself, we recommend you to set it in user setting only, not using workspace and folder level.

![image](https://user-images.githubusercontent.com/14052197/104865030-c66cd900-5975-11eb-9477-5296bb7627a0.png)
